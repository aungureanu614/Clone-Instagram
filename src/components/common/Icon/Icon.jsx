import React from 'react'
import PropTypes from 'prop-types'

const Icon = ({ image, alt, className }) => (
  <img src={image} alt={alt} className={className} />
)

Icon.propTypes = {
  image: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
}

Icon.defaultProps = {
  className: '',
}

export default Icon
