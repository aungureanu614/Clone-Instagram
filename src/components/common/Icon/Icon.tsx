import React from 'react'

interface IconOptions {
  image: string;
  alt: string;
  className: string;
}

const Icon = (icon: IconOptions) => (
  <img src={icon.image} alt={icon.alt} className={icon.className} />
)

export default Icon
