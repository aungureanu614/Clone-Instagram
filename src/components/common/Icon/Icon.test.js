import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './Icon'
import icon from '../../../images/ig-logo.png'

test('Icon renders', () => {
  const component = renderer.create(
    <Icon image={icon} alt="Instagram Icon" className="Icon" />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
