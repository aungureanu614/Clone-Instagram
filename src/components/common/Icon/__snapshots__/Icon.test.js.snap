// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders 1`] = `
<img
  alt="Instagram Icon"
  className="Icon"
  src="ig-logo.png"
/>
`;
